src/openvic-simulation/economy/ProductionType.cpp

```diff
@@ -29,96 +32,99 @@ bool ProductionType::parse_scripts(GameManager const& game_manager) {
-		ret &= cb(std::move(employed_pops));
+		std::vector<Job> jobs;
+		bool ret = expect_list(_expect_job(good_manager, pop_manager, vector_callback(jobs)))(node);
+		ret &= cb(std::move(jobs));
 		return ret;
 	};
 }
 
-#define POPTYPE_CHECK(employed_pop) \
-	if ((employed_pop.pop_type == nullptr && !employed_pop.artisan) || \
-		(employed_pop.pop_type != nullptr && employed_pop.artisan)) { \
-		Logger::error("Invalid pop type parsed for owner of production type ", identifier, "!"); \
-		return false; \
-	}
-
 bool ProductionTypeManager::add_production_type(
-	std::string_view identifier, EmployedPop owner, std::vector<EmployedPop> employees, ProductionType::type_t type,
-	Pop::pop_size_t workforce, Good::good_map_t&& input_goods, Good const* output_goods, fixed_point_t value,
-	std::vector<ProductionType::bonus_t>&& bonuses, Good::good_map_t&& efficiency, bool coastal, bool farm, bool mine
+	std::string_view identifier, Job owner, std::vector<Job> jobs, ProductionType::template_type_t template_type,
+	Pop::pop_size_t base_workforce_size, Good::good_map_t&& input_goods, Good const* output_goods, fixed_point_t base_output_quantity,
+	std::vector<ProductionType::bonus_t>&& bonuses, Good::good_map_t&& maintenance_requirements, bool is_coastal, bool is_farm, bool is_mine
 ) {
 	if (identifier.empty()) {
 		Logger::error("Invalid production type identifier - empty!");
 		return false;
 	}
 
-	if (workforce <= 0) {
-		Logger::error("Workforce for production type ", identifier, " was 0 or unset!");
+	if (base_workforce_size <= 0) {
+		Logger::error("Base workforce size ('workforce') for production type ", identifier, " was 0 or unset!");
 		return false;
 	}
 
-	if (value <= 0) {
-		Logger::error("Value for production type ", identifier, " was 0 or unset!");
+	if (base_output_quantity <= 0) {
+		Logger::error("Base output quantity ('value') for production type ", identifier, " was 0 or unset!");
 		return false;
 	}
 
-	POPTYPE_CHECK(owner)
-
-	for (EmployedPop const& ep : employees) {
-		POPTYPE_CHECK(ep)
-	}
-
 	if (output_goods == nullptr) {
 		Logger::error("Output good for production type ", identifier, " was null!");
 		return false;
 	}
+	
+	if(template_type == ProductionType::template_type_t::ARTISAN) {
+		if(owner.get_pop_type() != nullptr || !jobs.empty()) {
+			Logger::warning("Artisanal production types don't use owner and employees. Effects are ignored.");
+		}
+	}
+	else {
+		if(owner.get_pop_type() == nullptr) {
+			Logger::error("Production type ", identifier, " lacks owner or has an invalid pop type.");
+			return false;
+		}
+
+		if(jobs.empty()) {
+			Logger::error("Production type ", identifier, " lacks jobs ('employees').");
+			return false;
+		}
+
+		for (int i = 0; i < jobs.size(); i++) {
+			const Job job = jobs[i];
+			if(job.get_pop_type()== nullptr) {
+				Logger::error("Production type ", identifier, " has invalid pop type in employees[",i,"].");
+				return false;
+			}
+		}
+	}
```

Comment:
```suggestion
	if (template_type == ProductionType::template_type_t::ARTISAN) {
		if(owner.get_pop_type() != nullptr || !jobs.empty()) {
			Logger::warning("Artisanal production types don't use owner and employees. Effects are ignored.");
		}
	} else {
		if (owner.get_pop_type() == nullptr) {
			Logger::error("Production type ", identifier, " lacks owner or has an invalid pop type.");
			return false;
		}

		if (jobs.empty()) {
			Logger::error("Production type ", identifier, " lacks jobs ('employees').");
			return false;
		}

		for (size_t i = 0; i < jobs.size(); i++) {
			if (jobs[i].get_pop_type() == nullptr) {
				Logger::error("Production type ", identifier, " has invalid pop type in employees[",i,"].");
				return false;
			}
		}
	}
```
Minor formatting tweaks, also:
- `jobs.size()` is a `size_t`, so might as well use that type for `i`.
- `const Job job = jobs[i];` creates a copy of the `Job` object, you could instead create a const reference `Job const& job = jobs[i];` which doesn't require a copy or just use `jobs[i]` directly since it's only needed once.